src/Sylius/Component/Core/Model/Order.php

```diff
@@ -407,16 +363,12 @@ public function getPromotionCoupons()
     /**
      * {@inheritdoc}
      */
-    public function addPromotionCoupon($coupon)
+    public function addPromotionCoupon(BaseCouponInterface $coupon = null)
```

Comment:
Unfortunately, we must allow ``null`` value :/ After adding missing typehint, it throws exception if null is passed to function and it somehow happens when we type wrong coupon on cart summary page. It should be obviously changed, but not in this PR.